server/main_server/src/main.cpp

```diff
@@ -1449,6 +1459,47 @@ Environment Variables:
         };
     } // remove_leftover_agent_info_files_for_ips
 
+    auto apply_access_time_updates() -> void
+    {
+        try {
+            log_server::trace("{}: Applying access time updates.", __func__);
+
+            irods::experimental::client_connection conn;
+            irods::access_time_manager::access_time_data data;
+            DataObjInfo info{};
+            KeyValPair kvp{};
+            modDataObjMeta_t input{};
+            input.dataObjInfo = &info;
+            input.regParam = &kvp;
+
+            for (int i = 0; i < 250; ++i) {
```

Comment:
what is 250?

Reply:
That is temporary and will be replaced by a system which limits the processing of queued atime updates to a small time slice - e.g. 1500 milliseconds. The time slice approach is to keep the main server process responsive to signals, but also allow the processing of atime updates.

Reply:
rather than a separate child process to clear/process the queue?   then it's never blocking?

Reply:
I chose to avoid a new child process because it felt like overkill for such a simple thing. That approach also means we have to deal with more signaling. The current solution is simple and keeps the number of child processes to 2.

If we discover that this implementation cannot keep up, then that is a vote to switch to a more complicated design. I don't feel that will be the case. There are a few things this implementation still needs to do before it's finished though (e.g. ignore consecutive updates to a replica).

Reply:
Gotcha - so time slice makes sense to me too.   Resolving.